extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/v2Commands.scala

```diff
@@ -0,0 +1,346 @@
+import org.apache.kyuubi.plugin.spark.authz.util.AuthZUtils._
+
+object V2CommandType extends Enumeration {
+  type V2CommandType = Value
+
+  // traits' name from Spark v2commands and v2AlterTableCommands
+  val V2CreateTablePlan, V2WriteCommand, V2AlterTableCommand, V2PartitionCommand = Value
+
+  // custom types
+  val HasQuery, V2DdlTableCommand = Value
+}
+
+object v2Commands extends Enumeration {
+
+  import scala.language.implicitConversions
+
+  implicit def valueToV2CommandBuilder(x: Value): V2Command =
+    x.asInstanceOf[V2Command]
+
+  def accept(commandName: String): Boolean = {
+    try {
+      val cmd = v2Commands.withName(commandName)
+
+      // check spark version requirements
+      def passSparkVersionCheck: Boolean =
+        (StringUtils.isBlank(cmd.mostVer) || isSparkVersionAtMost(cmd.mostVer)) &&
+          (StringUtils.isBlank(cmd.leastVer) || isSparkVersionAtLeast(cmd.leastVer))
+
+      passSparkVersionCheck && cmd.enabled
+    } catch {
+      case _: NoSuchElementException => false
+    }
+  }
+
+  val defaultBuildInput: (LogicalPlan, ArrayBuffer[PrivilegeObject], Seq[V2CommandType]) => Unit =
+    (p, inputObjs, cmdTypes) => {
+      cmdTypes.foreach {
+        case HasQuery =>
+          val query = getFieldVal[LogicalPlan](p, "query")
+          buildQuery(query, inputObjs)
+        case _ =>
+      }
+    }
+
+  val defaultBuildOutput: (
+      LogicalPlan,
+      ArrayBuffer[PrivilegeObject],
+      Seq[V2CommandType],
+      PrivilegeObjectActionType) => Unit =
+    (p, outputObjs, cmdTypes, outputObjsActionType) => {
+      cmdTypes.foreach {
+        case V2CreateTablePlan =>
+          val table = invoke(p, "tableName").asInstanceOf[Identifier]
+          outputObjs += v2TablePrivileges(table)
+
+        case V2WriteCommand =>
+          val table = getFieldVal[AnyRef](p, "table")
+          val tableIdent = getFieldVal[Option[Identifier]](table, "identifier")
+          if (tableIdent.isDefined) {
+            outputObjs += v2TablePrivileges(tableIdent.get, actionType = outputObjsActionType)
+          }
+
+        case V2AlterTableCommand if isSparkVersionAtLeast("3.2") =>
+          val table = getFieldVal[LogicalPlan](p, "table")
+          val tableIdent = getFieldVal[Identifier](table, "identifier")
+          outputObjs += v2TablePrivileges(tableIdent)
+
+        case V2DdlTableCommand =>
+          val table = getFieldVal[AnyRef](p, "child")
+          val tableIdent = getFieldVal[Identifier](table, "identifier")
+          outputObjs += v2TablePrivileges(tableIdent)
+
+        case V2PartitionCommand =>
+          val table = getFieldVal[AnyRef](p, "table")
+          val tableIdent = getFieldVal[Identifier](table, "identifier")
+          outputObjs += v2TablePrivileges(tableIdent)
+
+        case _ =>
+      }
+    }
+
+  case class V2Command(
```

Comment:
CommandConverter/CommandDetail/CommandInfo etc

Reply:
changed to `CmdPrivilegeBuilder`